lib/human.py

```diff
@@ -0,0 +1,146 @@
+        return node[':rhs-val']
+    else:
+        return pretty_indicator(node[":rhs-fn"], node[":rhs-val"], node.get(":rhs-window-days"))
+
+
+def pretty_condition(node) -> str:
+    return f"{pretty_lhs(node)} {pretty_comparison(node[':comparator'])} {pretty_rhs(node)}"
+
+
+def pretty_selector(node) -> str:
+    if node[":select-fn"] == ":bottom":
+        return f"bottom {node[':select-n']}"
+    elif node[":select-fn"] == ":top":
+        return f"top {node[':select-n']}"
+    else:
+        return f"UNEXPECTED :select-fn {node[':select-fn']}"
+
+
+def pretty_filter(node) -> str:
+    return f"{pretty_selector(node)} by {pretty_indicator(node[':sort-by-fn'], '____', node[':sort-by-window-days'])}"
+
+
+def print_children(node, depth=0, parent_node_branch_state: typing.Optional[logic.NodeBranchState] = None, file=None):
+    """
+    Recursively visits every child node (depth-first)
+    and pretty-prints it out.
+    """
+    if not parent_node_branch_state:
+        # current node is :root, there is no higher node
+        parent_node_branch_state = logic.NodeBranchState(1, [], "")
+    parent_node_branch_state = typing.cast(
+        logic.NodeBranchState, parent_node_branch_state)
+
+    current_node_branch_state = logic.advance_branch_state(
+        parent_node_branch_state, node)
+
+    def pretty_log(message: str):
+        s = "  " * depth
+
+        weight_factor = logic.extract_weight_factor(
+            parent_node_branch_state, node)
+        if not logic.is_weight_node(node) and weight_factor != 1:
+            s += f"{weight_factor:.0%} => "
+
+        s += message
+
+        if logic.is_asset_node(node):
+            s += f" (max: {current_node_branch_state.weight:.1%})"
+        print(s, file=file)
+
+    if logic.is_root_node(node):
+        pretty_log(node[":name"])
+    elif logic.is_equal_weight_node(node):
+        # sometimes children will have :weight (sometimes inserted as a no-op)
+        pretty_log("Weight equally:")
+    elif logic.is_specified_weight_node(node):
+        # children will have :weight
+        # ':weight': {':num': 88, ':den': 100}, (numerator and denominator)
+        pretty_log("Weight accordingly:")
+    elif logic.is_if_node(node):
+        pretty_log("if")
+    elif logic.is_if_child_node(node):
+        if not logic.is_conditional_node(node):
+            pretty_log("else")
+        else:
+            pretty_log(f"({pretty_condition(node)})")
+    elif logic.is_group_node(node):
+        pretty_log(f"// {node[':name']}")
+    elif logic.is_filter_node(node):
+        pretty_log(pretty_filter(node))
+    elif logic.is_asset_node(node):
+        pretty_log(logic.get_ticker_of_asset_node(node))
+    else:
+        pretty_log(f"UNIMPLEMENTED: {node[':step']}")
+
+    for child in logic.get_node_children(node):
+        print_children(child, depth=depth+1,
+                       parent_node_branch_state=current_node_branch_state, file=file)
+
+
+def convert_to_pretty_format(root_node) -> str:
```

Comment:
The main entry point of this transpiler; the other functions give one an idea of what each transpiler needs to be able to express and traverse.